test/factories/host.rb

```diff
@@ -4,35 +4,73 @@
   # This is necessary to convince spring that the class is not defined elsewhere
   Object.send(:remove_const, :WHost) if defined?(WHost)
 
+  SSGS = SupportedSsg.all
+
   # A writable version of the Host model for the factory only
   class WHost < Host
-    self.table_name = 'inventory.hosts_v1_1'
+    if Rails.env.test?
+      self.table_name = 'inventory.hosts_v1_1'
+    elsif Rails.env.development?
+      self.table_name = 'hosts'
+
+      establish_connection(
+        Rails.configuration.database_configuration[Rails.env].merge(
+          'database' => 'insights'
+        )
+      )
+    end
 
     def readonly?
       false
     end
   end
 
+  sequence(:account_number) { |n| format('%05<num>d', num: n) }
+
   factory :host, class: WHost do
     id { SecureRandom.uuid }
-    account { User.current&.account&.account_number }
+    account do
+      User.current&.account&.account_number || generate(:account_number)
+    end
     display_name { Faker::Lorem.sentence }
     tags { {} }
-    created { Time.zone.now }
-    updated { Time.zone.now }
     stale_timestamp { 10.years.since(Time.zone.now) }
-    system_profile do
-      {
-        'operating_system' => {
-          'major' => os_major_version,
-          'minor' => os_minor_version
+    if Rails.env.test?
+      created { Time.zone.now }
+      updated { Time.zone.now }
+      system_profile do
+        system_profile_data
+      end
+    elsif Rails.env.development?
+      reporter { 'puptoo' }
+      modified_on { Time.zone.now }
+      facts { {} }
+      canonical_facts do
+        {
+          'fqdn' => Faker::Lorem.sentence
         }
-      }
+      end
+
+      system_profile_facts do
+        system_profile_data
+      end
     end
 
     transient do
-      os_major_version { 7 }
-      os_minor_version { 9 }
+      os_version do
+        ssg = SSGS.sample
+        [ssg.os_major_version.to_i, ssg.os_minor_version.to_i]
+      end
+      os_major_version { os_version[0] }
+      os_minor_version { os_version[1] }
```

Comment:
I also decided to go ahead and make the OS dynamic based on what we have defined in supported_ssgs.yaml, since we often need different OS versions to test with (i.e. RHEL 6)